matrix/lambdas/daemons/worker.py

```diff
@@ -43,24 +46,31 @@ def run(self, format: str, worker_chunk_spec: typing.List[dict]):
         logger.debug(f"Worker running with parameters: worker_chunk_spec={worker_chunk_spec}, format={format}")
         # TO DO pass in the parameters in worker chunk spec flat
         self._parse_worker_chunk_spec(worker_chunk_spec)
+        num_bundles = len(self._bundle_uuids)
         exp_dfs = []
         qc_dfs = []
-        num_bundles = len(self._bundle_uuids)
-        for chunk_idx in range(num_bundles):
-            dss_zarr_store = DSSZarrStore(bundle_uuid=self._bundle_uuids[chunk_idx],
-                                          bundle_version=self._bundle_versions[chunk_idx],
-                                          dss_instance=self._deployment_stage)
-            group = zarr.group(store=dss_zarr_store)
-            exp_df, qc_df = convert_dss_zarr_root_to_subset_pandas_dfs(
-                group, self._input_start_rows[chunk_idx], self._input_end_rows[chunk_idx])
-            exp_dfs.append(exp_df)
-            qc_dfs.append(qc_df)
-
-            # log every tertile of bundles read
-            if any(chunk_idx == int(math.ceil((num_bundles - 1) * ((i + 1) / 3))) for i in range(2)):
-                logger.debug(f"{chunk_idx + 1} of {len(self._bundle_uuids)} bundles successfully read from the DSS")
-
-        # In some test cases, dataframes aren't actually returned. Don't try to
+
+        # Parallelize high latency bundle reads from DSS
+        with concurrent.futures.ThreadPoolExecutor(max_workers=25) as executor:
+            future_to_chunk_map = {executor.submit(self._parse_chunk_to_dataframe, chunk_idx): chunk_idx
+                                   for chunk_idx in range(num_bundles)}
+            for future in concurrent.futures.as_completed(future_to_chunk_map):
+                chunk_idx = future_to_chunk_map[future]
+                try:
+                    exp_df, qc_df = future.result()
+                except Exception as e:
+                    logger.debug(f"Parsing bundle uuid {self._bundle_uuids[chunk_idx]} from DSS "
+                                 f"to pandas.DataFrame caused exception {e}")
+                    logger.debug(traceback.format_exc())
```

Comment:
Do we want to keep this? I guess it could be helpful, but might flood logs. How often does this Exception get thrown now by the way? It was previously thrown ~30-50% of the time due to shared sessions in the DSSClient

Reply:
Should never happen now. Every exception should be interesting.

Reply:
Cool cool. Do you know if re raising the exception would be better? I'm only using traceback here to retrieve the stack trace

Reply:
Oh, yeah probably better to just `raise` after this.